core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -66,19 +68,64 @@
 
   private ConfigurationUtils() {} // prevent instantiation
 
+  /**
+   * Gets the embedded journal addresses to use for the given service type (either master-raft or
+   * job-master-raft).
+   *
+   * @param conf configuration
+   * @param serviceType the service to get addresses for
+   * @return the addresses
+   */
+  public static List<InetSocketAddress> getEmbeddedJournalAddresses(AlluxioConfiguration conf,
```

Comment:
This function is a bit unintuitive, perhaps we should split into `getJobMaster` and `getMaster` similar to the rpc address methods?

Reply:
Agreed, it got pretty out of hand trying to merge the two logics. Much nicer now as two separate methods